.github/workflows/ci.yml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  pull_request:
+    branches:
+      - "*"
+  schedule:
+    # Daily at 06:14
+    - cron: '14 6 * * *'
+
+jobs:
+  test:
+    name: ${{ matrix.os.name }} ${{ matrix.python.name }}
+    runs-on: ${{ matrix.os.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - name: Linux
+            runs-on: ubuntu-latest
+            matrix: linux
+          - name: Windows
+            runs-on: windows-latest
+            matrix: windows
+          - name: macOS
+            runs-on: macos-latest
+            matrix: macos
+        python:
+          - name: CPython 3.5
+            tox: py35
+            action: 3.5
+          - name: CPython 3.6
+            tox: py36
+            action: 3.6
+          - name: CPython 3.7
+            tox: py37
+            action: 3.7
+          - name: CPython 3.8
+            tox: py38
+            action: 3.8
+          - name: CPython 3.9
+            tox: py39
+            action: 3.9
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up ${{ matrix.python.name }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: '${{ matrix.python.action }}.0-alpha - ${{ matrix.python.action }}.X'
+          architecture: x64
+      - uses: twisted/python-info-action@v1.0.1
+      - name: Install Linux Qt5 dependencies
+        if: matrix.os.matrix == 'linux'
+        run: |
+          sudo apt-get update --yes
+          sudo apt-get install --yes libgl1
+      - name: Install
+        run: |
+          pip install --upgrade pip setuptools wheel
+          pip install tox
+      - name: Test
+        run: |
+          tox -v -e ${{ matrix.python.tox }}
+
+  all:
+    name: All
+    runs-on: ubuntu-latest
+    needs:
+      - test
+    steps:
+      - name: This
+        shell: python
+        run: |
+          import this
```

Comment:
What's the purpose of this?

Reply:
When you tell GitHub what checks are required you have to check individual jobs, not the workflow.  This All job gives you a single job to select there.  For example, if we add another python and don't do this then we have to go check three more jobs and do so in a way that will make other PRs fail until that change is merged into them etc.  With this it is (more) contained in the repo.

The `import this` is just a fast thing to put in the step that might occasionally make someone smile.

Reply:
Not sure I understand why an `All` job is important? If we add another Python, wouldn't we just add it to the build matrix?

Reply:
If that's all you do then it won't be marked as required and you will be allowed to merge a PR even when it fails.  The images below are from QTrio but show how I don't have to check every single job created by the matrix and keep those checks in sync with (one of the branches of) the code.  Note how the `All` job `needs:` the `test` job and thus is only successful (only even runs) if every matrixed 'sub-job' of `test` succeeds.

The alternative is that when you change the matrix (either addition or subtraction of sub-jobs (cells?  I don't know...)) you also go into the GitHub project settings and change the required jobs.  One issue is that you have to do this at all.  The other issue is that when you do that then all other open PR's are failing.  It is a global setting despite the CI config and the actually defined jobs being a branch setting.

This is certainly not required, most projects do not do this.  But, mine do.  `:]`  In this case we don't control the required jobs anyways until we get PyQt5-stubs moved into an org, so it kind of doesn't matter until then.  We're just going to have to look for ourselves before merging regardless.

![image](https://user-images.githubusercontent.com/543719/102843090-77637080-43d6-11eb-99fc-274677cd7102.png)
![image](https://user-images.githubusercontent.com/543719/102843111-85b18c80-43d6-11eb-9282-3d050f0f8169.png)
![image](https://user-images.githubusercontent.com/543719/102843171-a37ef180-43d6-11eb-8537-23c1c9b88845.png)

Reply:
Sounds good to me